### ISSUE TITLE
How to add js code in ViewComponent

### ISSUE THREAD

**Issue by IrakliLomidze** [aspnet/Docs#10609]
1) How to add js code in ViewComponent
2) How to property reference on JQuery
3) What is  $( document ).ready(function() Alternative in ViewComponent Lifecylce
4) How to avoid multiple js code loading, when use ViewComponents

Please Put all of this in to the current document
Thank you in Advance.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9d2902f4-b6fd-ba1c-6762-d7c26f5eb4b9
* Version Independent ID: b5715d72-5f10-bb93-1051-7f06dfa9cde8
* Content: [View components in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-components?view=aspnetcore-2.2)
* Content Source: [aspnetcore/mvc/views/view-components.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/views/view-components.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@guardrex 	isn't Blazer more appropriate for this request?

**Comment by guardrex** [comment]
1. I would've thought a section would work. I haven't tried it, but it should 🤞 be ok (note that script file loading like this can block rendering).

   ```cshtml
   @section Scripts {
       <script type="text/javascript" src="/scripts/my-script.js"></script>
       <script>
         // JavaScript
       </script>
   }
   ```

2. I don't understand the 2nd question.
3. The `$(document).ready(function() { ... });` or `$(function() { ... })` executes when the DOM is ready. There's no alternative.
4. Load the common code outside of the VC.

These behaviors probably aren't different than placing JS into a View/RP page generally. If the `@section` works, I don't know that there's anything else to document, per se.

**Comment by AlbertoPa** [comment]
There is a discussion here on this topic: https://github.com/aspnet/AspNetCore/issues/2037

There are some peculiarities (e.g. scripts not rendered in the shared _Layout, need to load JS library in the VC). At some point in another issue it was said that a cleaner solution would be introduced with Razor Components in .Net 3 (assuming one wants the extra layers of SignalR and wants to deal with different implementation strategies just for a view component).

**Comment by guardrex** [comment]
Thanks @AlbertoPa for clearing that up.

I found that remark at https://docs.microsoft.com/aspnet/core/mvc/views/layout#sections ...

> Sections defined in a page or view are available only in its immediate layout page. They cannot be referenced from partials, view components, or other parts of the view system.

We can close here because VC behavior is already documented. The engineers seem to have this under consideration for a future release. @AlbertoPa is correct about Razor Components: It's not available until 3.0. I'm not aware of how `BlazorLayoutComponent` will behave in relation to a Razor view layout, if that becomes a valid approach. I'll keep it in mind as we work the Razor Components docs over here for 3.0.

**Comment by mehdipanjwani-ains** [comment]
Specifying another layout for viewcomponents with scripts will duplicate scripts. And @section is just ignored during rendering. So I replaced jquery document ready with plain javascript domcontentloaded and any jquery can be written inside domcontentloaded. Not a permanent good approach but works for me.

```
<script>
    document.addEventListener('DOMContentLoaded', function (event) {
    console.log($ === jQuery)
});
</script>
```
